### PR TITLE
Use tag names in all user-agent selectors.

### LINKED ISSUE #11004: Style sharing is disabled due to an outline rule
`:focus { outline: thin dotted; } /* FIXME: 'outline: auto' ? */` in `user-agent.css` is disabling all style sharing. This is obviously bad for competitive performance.

### PR DESCRIPTION
**Pull request by notriddle**
Fixes #11004

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11022)

### CODE REVIEW

File: resources/user-agent.css [hunk 1]
```diff
@@ -87,13 +87,22 @@ sub, sup { line-height: normal; font-size: smaller; }
...
+a:link, a:visited,
+area:link, area:visited { text-decoration: underline; cursor: pointer; }
 a:link[rel~=help], a:visited[rel~=help],
 area:link[rel~=help], area:visited[rel~=help] { cursor: help; }
 
-:focus { outline: thin dotted; }  /* FIXME: 'outline: auto' ? */
+/*
+ * FIXME: use `outline: auto;`
+ */
+a:focus, area:focus { outline: thin dotted; }
```
Comment (jdm) [inline resources/user-agent.css]: This should cover input and button elements too.

**Comment by nox** [comment]
Is there no way to do that differently? Isn't that stylesheet directly taken from the HTML spec? And as far as I know, `<link>` can also match `:link`.

**Comment by notriddle** [comment]
The best long-term fix is to allow style sharing on elements that have non-traversal style-sharing. See [rust-selectors/src/matching.rs](https://github.com/notriddle/rust-selectors/blob/0635af9fdd121105dec0d5487903a97641fd96e3/src/matching.rs#L685-L688).

**Comment by pcwalton** [comment]
Could you file a follow up issue to do this properly? I'm OK with this for now, but I'd like to make sure we don't forget to implement this correctly.

**Comment by notriddle** [comment]
#11071

**Pull request merged**